sdk/src/lib.rs

```diff
@@ -57,13 +57,16 @@ impl ProverClient {
     #[allow(clippy::new_without_default)]
     pub fn new() -> Self {
         dotenv::dotenv().ok();
-        let private_key = env::var("PRIVATE_KEY").unwrap_or_default();
-        if private_key.is_empty() {
-            Self { client: None }
-        } else {
+        let remote_proving = env::var("REMOTE_PROVING").unwrap_or_else(|_| String::from("false"));
+
+        if remote_proving == "true" {
+            let private_key =
+                env::var("PRIVATE_KEY").unwrap_or_else(|_| panic!("PRIVATE_KEY is not set"));
```

Comment:
nit: `PRIVATE_KEY` must be set for remote proving.

Reply:
updated